File: rest-api/src/main/java/com/stats/aggregator/config/DatabaseConfig.java
package com.stats.aggregator.config;

import com.mongodb.Mongo;
import com.mongodb.MongoClient;
import com.stats.aggregator.common.utils.PropertiesHelper;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.config.AbstractMongoConfiguration;
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;


@Configuration
@EnableMongoRepositories("com.stats.aggregator.repositories")
public class DatabaseConfig extends AbstractMongoConfiguration {

    @Override
    protected String getDatabaseName() {
        return "prize-stats-aggregator";
    }

    @Override
    public Mongo mongo() throws Exception {
        Mongo mongo = new MongoClient(PropertiesHelper.getProperty("mongoDb.bindIp"),
                Integer.parseInt(PropertiesHelper.getProperty("mongoDb.port")));
        return mongo;
    }
}

File: scheduler/src/main/scala/com/stats/aggregator/scheduler/config/DatabaseConfig.scala
package com.stats.aggregator.scheduler.config

import com.mongodb.{Mongo, MongoClient}
import com.stats.aggregator.common.utils.PropertiesHelper
import org.springframework.context.annotation.Configuration
import org.springframework.data.mongodb.config.AbstractMongoConfiguration
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories

@Configuration
@EnableMongoRepositories(Array("com.stats.aggregator.scheduler.repo"))
class DatabaseConfig extends AbstractMongoConfiguration {
  override protected def getDatabaseName = "prize-stats-aggregator"

  @throws[Exception]
  override def mongo: Mongo = {
    new MongoClient(PropertiesHelper.getProperty("mongoDb.bindIp"), PropertiesHelper.getProperty("mongoDb.port").toInt)
  }
}
